cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -876,25 +875,39 @@ public void getDeployedAppSummary(HttpRequest request, HttpResponder responder,
       @QueryParam("format") @DefaultValue("markdown") String format)
       throws BadRequestException, NotFoundException, IOException {
     validateApplicationId(namespaceId, appName);
-    String summary = applicationLifecycleService.summarizeApp(namespaceId, appName, format);
+    ApplicationReference appRef = new ApplicationReference(namespaceId, appName);
+    ApplicationDetail appDetail = applicationLifecycleService.getLatestAppDetail(appRef, false);
+    AppRequest appRequest = new AppRequest<>(appDetail.getArtifact(), appDetail.getConfiguration());
+    String summary = applicationLifecycleService.summarizeApp(appRequest, format);
     responder.sendString(HttpResponseStatus.OK, summary);
   }
 
   /**
    * Endpoint to summarize drafted applications.
    *
-   * @param request The HTTP request object.
+   * @param request The FullHTTP request object.
    * @param responder The HTTP responder object.
    * @param namespaceId The namespace ID where the application is drafted.
    * @param format The format in which the summary should be returned (default is markdown).
-   * @throws NotImplementedException Thrown if the method is not implemented yet.
    */
   @POST
   @Path("/apps/summarize")
-  public void getAppSummary(HttpRequest request, HttpResponder responder,
+  public void getAppSummary(FullHttpRequest request, HttpResponder responder,
       @PathParam("namespace-id") final String namespaceId,
-      @PathParam("format") @DefaultValue("markdown") String format)
-      throws NotImplementedException{
-      throw  new NotImplementedException("This api request is not implemented.");
+      @QueryParam("format") @DefaultValue("markdown") String format)
+      throws BadRequestException, IOException {
+    try (Reader reader = new InputStreamReader(new ByteBufInputStream(request.content()),
+        StandardCharsets.UTF_8)) {
+      AppRequest<?> appRequest = GSON.fromJson(reader, AppRequest.class);
+      String summary = applicationLifecycleService.summarizeApp(appRequest, format);
+      responder.sendString(HttpResponseStatus.OK, summary);
+    } catch (JsonSyntaxException e) {
+      LOG.error("Request body is invalid json", e);
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
+    } catch (IllegalArgumentException e) {
+      responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
+    } catch (IOException e){
```

Comment:
Since i am catching other Exceptions and throwing Bad_request exceptions for them,not catching IO Exception and trying to directly throw it, is giving compilation error.